pkg/hintrunner/zero/zerohint_others_test.go

```diff
@@ -58,6 +59,134 @@ func TestZeroHintOthers(t *testing.T) {
 				check: varValueInScopeEquals("n", *feltUint64(1)),
 			},
 		},
+		"FindElement": {
+			{
+				operanders: []*hintOperander{
+					{Name: "array_ptr", Kind: apRelative, Value: addr(6)},
+					{Name: "elm_size", Kind: apRelative, Value: feltUint64(0)},
+					{Name: "key", Kind: apRelative, Value: feltUint64(1)},
+					{Name: "index", Kind: uninitialized},
+					{Name: "n_elms", Kind: apRelative, Value: feltUint64(0)},
+				},
+				makeHinter: func(ctx *hintTestContext) hinter.Hinter {
+					return newFindElementHint(ctx.operanders["array_ptr"], ctx.operanders["elm_size"], ctx.operanders["key"], ctx.operanders["index"], ctx.operanders["n_elms"])
+				},
+				errCheck: errorTextContains(fmt.Sprintf("Invalid value for elm_size. Got: %v", 0)),
+			},
+			{
+				operanders: []*hintOperander{
+					{Name: "array_ptr", Kind: apRelative, Value: addr(9)},
+					{Name: "elm_size", Kind: apRelative, Value: feltUint64(1)},
+					{Name: "key", Kind: apRelative, Value: feltUint64(999)},
+					{Name: "index", Kind: uninitialized},
+					{Name: "n_elms", Kind: apRelative, Value: feltUint64(0)},
+					{Name: "array.0", Kind: apRelative, Value: feltUint64(100)},
+					{Name: "array.1", Kind: apRelative, Value: feltUint64(200)},
+					{Name: "array.2", Kind: apRelative, Value: feltUint64(300)},
+				},
+				ctxInit: func(ctx *hinter.HintRunnerContext) {
+					err := ctx.ScopeManager.AssignVariable("__find_element_index", uint64(0))
+					if err != nil {
+						t.Fatal(err)
+					}
+				},
+				makeHinter: func(ctx *hintTestContext) hinter.Hinter {
+					return newFindElementHint(ctx.operanders["array_ptr"], ctx.operanders["elm_size"], ctx.operanders["key"], ctx.operanders["index"], ctx.operanders["n_elms"])
+				},
+				errCheck: errorTextContains(fmt.Sprintf("Invalid index found in __find_element_index. index: %v, expected key %v, found key: %v", feltUint64(0), feltUint64(999), feltUint64(100))),
+			},
+			{
+				operanders: []*hintOperander{
+					{Name: "array_ptr", Kind: apRelative, Value: addr(9)},
+					{Name: "elm_size", Kind: apRelative, Value: feltUint64(1)},
+					{Name: "key", Kind: apRelative, Value: feltUint64(300)},
+					{Name: "index", Kind: uninitialized},
+					{Name: "n_elms", Kind: apRelative, Value: feltUint64(0)},
+					{Name: "array.0", Kind: apRelative, Value: feltUint64(100)},
+					{Name: "array.1", Kind: apRelative, Value: feltUint64(200)},
+					{Name: "array.2", Kind: apRelative, Value: feltUint64(300)},
+				},
+				ctxInit: func(ctx *hinter.HintRunnerContext) {
+					err := ctx.ScopeManager.AssignVariable("__find_element_index", uint64(2))
+					if err != nil {
+						t.Fatal(err)
+					}
+				},
+				makeHinter: func(ctx *hintTestContext) hinter.Hinter {
+					return newFindElementHint(ctx.operanders["array_ptr"], ctx.operanders["elm_size"], ctx.operanders["key"], ctx.operanders["index"], ctx.operanders["n_elms"])
+				},
+				check: func(t *testing.T, ctx *hintTestContext) {
+					varValueEquals("index", feltUint64(2))(t, ctx)
```

Comment:
this test should fail because `n_elms == 0`

Reply:
not exactly, in this case the `__find_element_index` is present in the scope, which means that the part of the code, where `n_elms` is utilised is not accessible.

Reply:
oh yeah i see, my bad